src/clusterfuzz/_internal/bot/tasks/utasks/progression_task.py

```diff
@@ -405,6 +405,15 @@ def _set_regression_testcase_upload_url(
   if not fuzz_target:
     # No work to do, only applicable for engine fuzzers.
     return
+
+  if testcase.uploader_email:
+    logs.log_warn('Not saving uploaded to regression corpus.')
+    return
+  upload_metadata = data_types.TestcaseUploadMetadata.query(
+      data_types.TestcaseUploadMetadata.testcase_id == testcase.key.id()).get()
+  if upload_metadata:
```

Comment:
Could you please add a comment explaining what the presence of `upload_metadata` means and why it leads to skippping the upload?